common/components/Header/components/CustomNodeModal.tsx

```diff
@@ -0,0 +1,260 @@
+                      checked={this.state.customEip155}
+                      onChange={this.handleCheckbox}
+                    />
+                    {' '}
+                    <span>Supports EIP-155</span>
+                  </label>
+                </div>
+              </div>
+            }
+
+            <div className="row">
+              <div className="col-sm-9">
+                <label>URL</label>
+                {createInput("url", "http://127.0.0.1/")}
+              </div>
+
+              <div className="col-sm-3">
+                <label>{translate('NODE_Port')}</label>
+                {createInput("port", "8545", "number")}
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-sm-12">
+                <label>
+                  <input
+                    type="checkbox"
+                    name="hasAuth"
+                    checked={this.state.hasAuth}
+                    onChange={this.handleCheckbox}
+                  />
+                  {' '}
+                  <span>HTTP Basic Authentication</span>
+                </label>
+              </div>
+            </div>
+            {this.state.hasAuth &&
+              <div className="row">
+                <div className="col-sm-6">
+                  <label>Username</label>
+                  {createInput("username")}
+                </div>
+                <div className="col-sm-6">
+                  <label>Password</label>
+                  {createInput("password", undefined, "password")}
+                </div>
+              </div>
+            }
+          </form>
+        </div>
+      </Modal>
+    );
+  }
+
+  private getInvalids(): { [key: string]: boolean } {
+    const {
+      network,
+      url,
+      port,
+      hasAuth,
+      username,
+      password,
+      customEip155,
+      customChainId,
+    } = this.state;
+    const required = ["name", "url", "port", "network"];
+    const invalids: { [key: string]: boolean } = {};
+
+    // Required fields
+    required.forEach((field) => {
+      if (!this.state[field]) {
+        invalids[field] = true;
+      }
+    });
+
+    // Somewhat valid URL, not 100% fool-proof
+    if (!/https?\:\/\/\w+/i.test(url)) {
+      invalids.url = true;
+    }
+
+    // Numeric port within range
+    const iport = parseInt(port, 10);
+    if (!iport || iport < 1 || iport > 65535) {
+      invalids.port = true;
+    }
+
+    // If they have auth, make sure it's provided
+    if (hasAuth) {
+      if (!username) {
+        invalids.username = true;
+      }
+      if (!password) {
+        invalids.password = true;
+      }
+    }
+
+    // Numeric chain id, if provided
+    if (network === "custom" && customEip155 && !parseInt(customChainId, 10)) {
+      invalids.customChainId = true;
+    }
+
+    return invalids;
+  }
+
+  private handleChange = (ev: React.SyntheticEvent<
+    HTMLInputElement | HTMLSelectElement
```

Comment:
Instead, you can do `React.FormEvent<HTMLInputElement | HTMLSelectElement>`,
Then when you access `ev.currentTarget.value` you dont need to typecast anything